pulsar-transaction/common/src/main/java/org/apache/pulsar/transaction/common/TxnStatus.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.transaction.common;
+
+/**
+ * A enum represents the status of a transaction.
+ */
+public enum TxnStatus {
+
+    // A new transaction is open.
+    OPEN,
+    // A transaction is in the progress of committing.
+    COMMITTING,
+    // A transaction is already committed.
+    COMMITTED,
+    // A transaction is in the progress of aborting.
+    ABORTING,
+    // A transaction is already aborted.
+    ABORTED;
+
+    /**
+     * Check if the a status can be transaction to a new status.
+     *
+     * @param newStatus the new status
+     * @return true if the current status can be transitioning to.
+     */
+    public boolean canTransitionTo(TxnStatus newStatus) {
+        TxnStatus currentStatus = this;
+
+        switch (currentStatus) {
```

Comment:
I think this would be clearer if all assertions were positive (i.e. you can only transition to X if you are Y)

Reply:
I am treating this as a coding preference comment. If there is no fundamental issues, I will keep it as it is.